src/core/os.h

```diff
@@ -20,6 +20,8 @@ extern "C" {
 #ifdef _WIN32
 #include "windows/include/unistd.h"
 #include "windows/include/platform.h"
+#else
+#include <unistd.h>
```

Comment:
Since it's included in this header now, perhaps you could remove the includes of ```unistd.h``` from the files, which already include ```os.h``` so that it wouldn't be included twice. From what I've seen these are ```out.c```, ```os_posix.c``` and ```util.c```.